fe/fe-core/src/main/java/com/starrocks/qe/ConnectContext.java

```diff
@@ -109,6 +114,8 @@ public class ConnectContext {
     protected MysqlSerializer serializer;
     // Variables belong to this session.
     protected SessionVariable sessionVariable;
+    // all the modified session variables, will forward to master
+    protected List<SetVar> modifiedSessionVariables = new ArrayList<>();
```

Comment:
List may be very long if set one variable multi times. Is it better to use a map to record the modified session varialbes?

Reply:
Using a map is complicated because it requires `hash()` & `compare()` to all the subclass of `Expr`. Besides, I think this list would be in the size of 20 at most, and wouldn't cost too much memory.

Reply:
why 20 at most?

Reply:
Done.